public/app/features/dashboard-scene/serialization/buildNewDashboardSaveModel.ts

```diff
@@ -62,3 +83,75 @@ export async function buildNewDashboardSaveModel(urlFolderUid?: string): Promise
 
   return data;
 }
+
+export async function buildNewDashboardSaveModelV2(
+  urlFolderUid?: string
+): Promise<DashboardWithAccessInfo<DashboardV2Spec>> {
+  let variablesList = defaultDashboardV2Spec().variables;
+
+  if (config.featureToggles.newDashboardWithFiltersAndGroupBy) {
+    // Add filter and group by variables if the datasource supports it
+    const defaultDs = await getDatasourceSrv().get();
+
+    if (defaultDs.getTagKeys) {
+      const datasourceRef = {
+        type: defaultDs.meta.id,
+        uid: defaultDs.uid,
+      };
+
+      const filterVariable: AdhocVariableKind = {
+        kind: 'AdhocVariable',
+        spec: { ...defaultAdhocVariableSpec(), name: 'Filter', datasource: datasourceRef },
+      };
+
+      const groupByVariable: GroupByVariableKind = {
+        kind: 'GroupByVariable',
+        spec: {
+          ...defaultGroupByVariableSpec(),
+          datasource: datasourceRef,
+          name: 'Group by',
+        },
+      };
+
+      variablesList = (variablesList || []).concat([filterVariable, groupByVariable]);
+    }
+  }
+
+  const data: DashboardWithAccessInfo<DashboardV2Spec> = {
+    apiVersion: 'v2alpha0',
+    kind: 'DashboardWithAccessInfo',
+    spec: {
+      ...defaultDashboardV2Spec(),
+      title: 'New dashboard',
+      timeSettings: {
+        ...defaultTimeSettingsSpec(),
+        timezone: config.bootData.user?.timezone || defaultTimeSettingsSpec().timezone,
+      },
+    },
+    access: {
+      canStar: false,
+      canShare: false,
+      canDelete: false,
+      // Not sure this should belong here or to metadata.annotations
+      isNew: true,
```

Comment:
I think it makes sense to have this in metadata.annotations

Reply:
it's gonna be moved to metadata.annotations, specifically expressed through client side annotations.  We have discussed this in this thread: https://raintank-corp.slack.com/archives/C07SB1G4BC1/p1733319636821879?thread_ts=1733312540.215769&cid=C07SB1G4BC1.